src/Components/Web/src/RendererRegistryEventDispatcher.cs

```diff
@@ -18,8 +18,7 @@ public static class RendererRegistryEventDispatcher
         /// For framework use only.
         /// </summary>
         [JSInvokable(nameof(DispatchEvent))]
```

Comment:
It looks like we now only call this through JS interop in the WASM case. We *could* technically make it not `[JSInvokable]` now, and make it `internal`, and have WASM invoke it directly, bypassing JS interop. That would certainly make it less convenient to misuse by people doing weird things to raise artificial events from JavaScript.

I'm not super fixated that we must do that since I know it will complicate parameter passing a bit. The code here would have to deserialize a JSON representation of `BrowserEventDescriptor` manually, which would be a bit of a distraction. So I'll leave it up to you whether to make this not publicly invokable or not!

Reply:
I'm open to doing it, but can we file a separate issue and to it in preview9? I don't want to extend the scope of this PR even more.

Reply:
Actually I wonder if it might be valuable enough that we really should do it. As it stands here, someone could use this in server-side Blazor to trigger events that bypass the normal logging system.

Maybe at the least we should have a guard clause in here that checks if the current platform is WebAssembly, and if not, throws. There's no supported use case for this method on server-side Blazor, is there?

Reply:
Ah I wrote my comment before I saw yours. Yes the actual refactoring could be done in a later PR, but would you be open to including the guard clause in this PR just in case we never get to doing the refactoring? Should be pretty trivial.

Reply:
> Maybe at the least we should have a guard clause in here that checks if the current platform is WebAssembly, and if not, throws. There's no supported use case for this method on server-side Blazor, is there?

Nope. I'm ok with having a guard to avoid dispatching it outside web-assembly.

Reply:
Great. You probably know about this already, but just in case not: https://github.com/aspnet/AspNetCore/blob/master/src/Components/Components/src/PlatformInfo.cs#L10